handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -310,7 +310,8 @@ private void replaceHandler(ChannelHandlerContext ctx, Selection selection) {
         this.selection = selection;
         try {
             sslEngine = selection.context.newEngine(ctx.alloc());
-            ctx.pipeline().replace(this, SslHandler.class.getName(), selection.context.newHandler(sslEngine));
+            SslHandler handler = SslContext.newHandler(sslEngine);
```

Comment:
@normanmaurer depends, if you look at the unit test. If I know that Future will complete no matter what I can remove some of the `try-finally` and it will also not matter if I call the `super.` before or after adding the listener.